doc/rpmbuild.8

```diff
@@ -256,6 +266,18 @@ options are currently set in
 \fIrpmrc\fR and
 \fImacros\fR
 configuration file(s).
+.SS "DYNAMIC BUILD REQUIREMENTS"
+.PP
+When the %generate_buildrequires stage is executed and some of the resulting
+requirements are not satisfied, \fBrpmbuild\fR exits with code 11 and does not
+create the source package.
+Instead, a package ending with \fIbuildreqs.nosrc.rpm\fR is created, which has
+all the build requirements, including the newly generated ones.
```

Comment:
This is not fully correct. It has `all the build requirements that were possible to generate at that point of time`, we need to point out for people that sometimes they will need to run `-br`, install deps and then run `-br` again and it might generate new dependencies.

Reply:
Oh, right. Thanks! I'll fix that.

Reply:
@ignatenkobrain, just to clarify, is that because of the fact that a `%generate_buildrequires` macro could be generated during the previous iteration?

Reply:
yes, that section is executed during each built, so if it has some check `test -x /usr/bin/foo ...`, then once foo will be available, it might generate new set of requires.

that's why mock does `rpmbuild -br` + `dnf builddep` until it stops generating any new dependencies.

Reply:
Ah, I see. The [dummy spec](https://github.com/rpm-software-management/rpm/files/3911787/dummy-pkg.spec.txt) that @hrnciar provided in the related issue actually does exactly that.

Reply:
Reworded, please let me know if it's better now.